urwid/widget.py

```diff
@@ -1412,7 +1414,7 @@ def _normalize_to_caption(self, text):
             return text
         if tu:
             return text.encode('ascii') # follow python2's implicit conversion
-        return text.decode('ascii')
+        return text.decode(sys.stdin.encoding)
```

Comment:
We do agree though that anything that is called by `set_text` is not a good place to handle this?

Perhaps `keypress` is not the right place to deal with encodings, but it seemed like a fair choice given that there is already some encoding going on there. I'm not really familiar with urwid's codebase and thought `keypress` was already fairly low-level.